server/services/event/getEventDetails.ts

```diff
@@ -118,21 +119,24 @@ export interface AdminEventGetResponse extends Pick<Event, typeof adminEventGetE
 
 export type EventGetResponseType<A extends boolean> = true extends A ? AdminEventGetResponse : EventGetResponse;
 
+/**
+ * @param slugOrId Event id if admin === true, slug if admin === false.
+ * @param admin Whether or not to return results for the admin view.
+ */
 export default async function getEventDetails<A extends boolean>(
-  id: number, admin: A,
+  slugOrId: string | number, admin: A,
```

Comment:
Not a fan of using single argument to mean two things in different scenarios. How about options-object pattern? 

```tsx
type GetEventDetailsOptions = {
  id?: string;
  slug?: string;
  admin: A;
}

// or if id or slug are required, maybe:

type GetEventDetailsOptions = {
  id: string;
  admin: A;
} | {
  slug: string;
  admin: A;
}

// Then function would have argument

fn(options: GetEventDetailsOptions)

// and called

fn({ slug: "dasdsa", admin: true })
```

Reply:
Will definitely fix. The `getEventDetails` function has definitely seen too much evolution in its current form.